apps/els_lsp/src/els_code_action_provider.erl

```diff
@@ -35,31 +35,35 @@ code_actions(Uri, _Range, #{<<"diagnostics">> := Diagnostics}) ->
   lists:flatten([make_code_action(Uri, D) || D <- Diagnostics]).
 
 -spec make_code_action(uri(), map()) -> [map()].
-make_code_action(Uri, #{<<"message">> := Message, <<"range">> := Range}) ->
+make_code_action(Uri,
+  #{<<"message">> := Message, <<"range">> := Range} = D) ->
+    Data = maps:get(<<"data">>, D, <<>>),
   make_code_action(
-    [ {"function (.*) is unused", fun action_export_function/3}
-    , {"variable '(.*)' is unused", fun action_ignore_variable/3}
-    , {"variable '(.*)' is unbound", fun action_suggest_variable/3}
+    [ {"function (.*) is unused", fun action_export_function/4}
```

Comment:
Good that we are now passing the optional data field to the specific functions. It could be a nice way to exchange information. Later on (no need to do it as part of this PR), we could always ensure there's data in a diagnostic when constructing one. We may also want to use a map instead of a binary and base64 encode it to ensure it can be passed around between client and server. See the `els_utils:base64_encode_term/1` function and its usages for Call Hierarchy. Again, this is more of a mental note and can be done as a separate task.